bids/analysis/analysis.py

```diff
@@ -226,7 +226,14 @@ def _merge_contrast_inputs(self, inputs):
 
         for name, contrasts in groups.items():
             # Create a DF with contrasts in rows and contrast names in columns
-            data = [{**c.entities, **{c.name: 1}} for c in contrasts]
+            data = []
+            for c in contrasts:
+                c_data = {**c.entities, **{c.name: 1}}
+                for ent in ent_cols:
+                    if ent not in c_data:
+                        c_data[ent] = 'n/a'
+                data.append(c_data)
```

Comment:
This feels a little clearer to me.

```suggestion
            baseline = {ent: "n/a" for ent in ent_cols}
            data = [{**baseline, **c.entities, **{c.name: 1}} for c in contrasts]
```

In practice, I'd put the `baseline` line outside the `for` loop, but a pain to suggest around a deleted line.